router/usecase/candidate_routes.go

```diff
@@ -247,9 +264,12 @@ func GetCandidateRoutesNew(poolsByDenom map[string][]sqsdomain.PoolI, tokenIn sd
 					continue
 				}
 
-				_, ok := poolsByDenom[denom]
-				if !ok {
-					logger.Debug("no pools found for denom in candidate route search", zap.String("denom", denom))
+				rankedPools, err := c.candidateRouteDataHolder.GetRankedPoolsByDenom(currenTokenInDenom)
+				if err != nil {
+					return sqsdomain.CandidateRoutes{}, err
+				}
+				if len(rankedPools) == 0 {
+					c.logger.Debug("no pools found for denom in candidate route search", zap.String("denom", denom))
```

Comment:
just a random idea, would it be any benefit feed this data to our observability service?

Reply:
Thanks for suggestion!. Specifically for this one - I don't think so. The reason is that this branch may be taken in the expected cases. It also happens quite frequently. Adding some counter/alert for this specific case would result in too much noise.

However, it might be helpful for debugging, thus, I added a log at the `Debug` level.